utils/crypto/bls/signer/rpcsigner/client.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package rpcsigner
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc"
+
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+
+	pb "github.com/ava-labs/avalanchego/proto/pb/signer"
+)
+
+var (
+	_                   bls.Signer = (*Client)(nil)
+	ErrorEmptyPublicKey            = errors.New("empty public key")
+	ErrorEmptySignature            = errors.New("empty signature")
+)
+
+type Client struct {
+	client pb.SignerClient
+	conn   *grpc.ClientConn
+	pk     *bls.PublicKey
+}
+
+func NewClient(conn *grpc.ClientConn) (*Client, error) {
+	client := pb.NewSignerClient(conn)
+
+	pubkeyResponse, err := client.PublicKey(context.TODO(), &pb.PublicKeyRequest{})
+	if err != nil {
+		conn.Close()
+		return nil, err
+	}
+
+	pkBytes := pubkeyResponse.GetPublicKey()
+
+	if pkBytes == nil {
+		conn.Close()
+		return nil, ErrorEmptyPublicKey
+	}
+
+	pk, err := bls.PublicKeyFromCompressedBytes(pkBytes)
+	if err != nil {
+		conn.Close()
+		return nil, err
+	}
+
+	return &Client{
+		client: client,
+		conn:   conn,
+		pk:     pk,
+	}, nil
+}
+
+func (c *Client) Close() error {
+	return c.conn.Close()
+}
```

Comment:
Related to my last [comment](https://github.com/ava-labs/avalanchego/pull/3696/files#r1949626374) but I think this can be removed as well